cocos/asset/assets/image-asset.jsb.ts

```diff
@@ -36,82 +38,54 @@ export type ImageAsset = JsbImageAsset;
 export const ImageAsset: typeof JsbImageAsset = jsb.ImageAsset;
 const jsbWindow = jsb.window;
 
-export interface IMemoryImageSource {
-    _data: ArrayBufferView | null;
-    _compressed: boolean;
-    width: number;
-    height: number;
-    format: number;
-    mipmapLevelDataSize?: number[];
-}
-
-export type ImageSource = HTMLCanvasElement | HTMLImageElement | IMemoryImageSource | ImageBitmap;
-
 const extnames = ['.png', '.jpg', '.jpeg', '.bmp', '.webp', '.pvr', '.pkm', '.astc'];
 
-function isImageBitmap (imageSource: any): boolean {
-    return !!(sys.hasFeature(sys.Feature.IMAGE_BITMAP) && imageSource instanceof ImageBitmap);
-}
-
-function isNativeImage (imageSource: ImageSource): imageSource is (HTMLImageElement | HTMLCanvasElement | ImageBitmap) {
-    if (ALIPAY || XIAOMI || BAIDU) {
-        // We're unable to grab the constructors of Alipay native image or canvas object.
-        return !('_data' in imageSource);
-    }
-    if (JSB && (imageSource as IMemoryImageSource)._compressed === true) {
-        return false;
-    }
-
-    return imageSource instanceof jsbWindow.HTMLImageElement || imageSource instanceof jsbWindow.HTMLCanvasElement || isImageBitmap(imageSource);
-}
-
 // TODO: we mark imageAssetProto as type of any, because here we have many dynamic injected property @dumganhar
 const imageAssetProto: any = ImageAsset.prototype;
 
-imageAssetProto._ctor = function (nativeAsset?: ImageSource) {
+imageAssetProto._ctor = function (nativeAsset?: ImageData | IMemoryImageSource | HTMLCanvasElement | HTMLImageElement | ImageBitmap) {
     jsb.Asset.prototype._ctor.apply(this, arguments);
     this._width = 0;
     this._height = 0;
-    this._nativeData = {
-        _data: null,
-        width: 0,
-        height: 0,
-        format: 0,
-        _compressed: false,
-        mipmapLevelDataSize:[],
-    };
-
-    if (nativeAsset !== undefined) {
-        this.reset(nativeAsset);
+    if (nativeAsset instanceof ImageData) {
+        this._imageData = nativeAsset;
+    } else {
+        this._imageData = new ImageData(nativeAsset);
     }
 };
 
 Object.defineProperty(imageAssetProto, '_nativeAsset', {
     configurable: true,
     enumerable: true,
     get () {
-        return this._nativeData;
+        return this._imageData.source;
     },
-    set (value: ImageSource) {
-        if (!(value instanceof jsbWindow.HTMLElement) && !isImageBitmap(value)) {
-            (value as IMemoryImageSource).format = (value as IMemoryImageSource).format || this.format;
```

Comment:
Was this line removed ? value.format doesn't need to be set to `this.format` now?

Reply:
This is implemented in the reset function.